### ISSUE TITLE
Cannot place order at checkout with Braintree and 3DS enabled using billing addresses without region set

### ISSUE THREAD

**Issue by dphilipps** [magento/magento2#34204]
### Preconditions (*)

1. Magento 2.4.3
2. Braintree enabled
3. Card Payments enabled
4. **3D Secure enabled**
5. **Use billing address without state/region set**

### Steps to reproduce (*)

1. Enable Braintree in Sandbox mode.
2. Sandbox credentials: 
PUBLIC: hpbfzdqypph3njjq
PRIVATE: 4de9983821adb24738636e502c557248
MERCHANTID: vq82jrtv65hsgjbk
2. Enable Braintree Card Payments
3. Enable Braintree 3D Secure
4. Sign in to front end as demo user
5. Under account edit default address and ensure set as default shipping and billing
6. Country: United Kingdom
7. Street address: 123 Somewhere
8. City: London
9. State/Province: [MUST BE LEFT EMPTY]
10. Postcode: SE3 9SZ
11. Make up any other address details
12. Save address
13. Add a product to basket and proceed to payment step
14. Select credit card
15. Enter card details:
4242 4242 4242 4242
12/23
123
16. Click 'Place Order'

### Expected result (*)

1. Order confirmation is displayed

### Actual result (*)

1. Cannot proceed past payment screen and **cannot place order**.  Button is non responsive and no errors are displayed on screen.  Console messages below:

![image](https://user-images.githubusercontent.com/44231639/135455925-5400f525-9d3b-41bf-89f7-c4e32fa918ba.png)

Note: If 3DS is disabled, then the same address (above) will complete OK.  Likewise, if a state/region is set it will too.

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ X] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

3DS is vital for processing credit card payments as is a regulatory requirement in some jurisdictions and some processors do not approve payments without, therefore disabling it is not a work around.  UK addresses which typically do not have state or region set will not be able to place orders with 3DS enabled.

**Comment by dphilipps** [comment]
@magento give me 2.4-develop instance

**Comment by dphilipps** [comment]
Hi @engcom-November 
How do i get an instance deployed with Braintree included in the build?  Even with country set in payment methods config, it does not appear.

Many thanks
Dave

**Comment by dphilipps** [comment]
I have verified these steps on a fresh composer install of M2.4.3

**Comment by IanFiretoys** [comment]
worth noting that while this seems to only occur using **saved addresses** for logged in users with a blank region, freshly entered ones work fine. (it was mentioned to save the address in the reproduction steps, but not explicitly anywhere else in the body)

also someone should include the JS error in this issue as text somewhere so it's not so hard for anyone else with the same issue to find. screenshots aren't searchable

```
hosted-fields.min.js:1 Error in callback function
hosted-fields.min.js:1 TypeError: Cannot read properties of null (reading 'length')
    at Object.validate (3d-secure.js:1)
    at validator-handler.js:1
    at Function.map (jquery.js:2)
    at Object.validate (validator-handler.js:1)
    at UiClass.handleNonce (cc-form.js:1)
    at UiClass.onPaymentMethodReceived (cc-form.js:1)
    at Object.<anonymous> (adapter.js:1)
    at hosted-fields.min.js:1
```

you're welcome, frantic googlers

anyway, thanks to @dphilipps  for logging this, and @nicholasscottfish for the fix, this has been a particularly nasty one to root out.

**Comment by nicholasscottfish** [comment]
> Patch supplied by Gene support directly in response to this issue, however I have not tested it yet.
> 
> [when region-code is not mandatory(for Magento 2.4.3).zip](https://github.com/magento/magento2/files/7399865/when.region-code.is.not.mandatory.for.Magento.2.4.3.zip)

I have updated my patch as it had a strict type check on null when it needed to be !=, to be safe I would use the official patch provided by @dphilipps as I have tested and this fixes the bug.

**Comment by engcom-Hotel** [comment]
Hi @dphilipps.

Thank you for reporting the issue.

We followed your steps and able to reproduce it on Magento 2.4-develop/2.4.3/2.4.3-p1. Hence confirming the issue.

Thank you

**Comment by engcom-Hotel** [comment]
Hello @dphilipps,

Fix for the issue is added to the 2.4.4-release of Braintree package.

With 'paypal/module-braintree' version 4.3.0 issue will be fixed.

Below is the patch which is added to 4.3.0 of 'paypal/module-braintree'

if (billingAddress.regionCode == null)

{       billingAddress.regionCode = undefined; }

We are closing this issue.

Thanks

**Comment by dphilipps** [comment]
Thank you @engcom-Hotel 

> [engcom-Hotel](/engcom-Hotel)
